src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java

```diff
@@ -30,81 +38,108 @@ public PostgresSQLConnection(DbConnection conn) {
 
   @Override
   public SQLConnection setAutoCommit(boolean autoCommit, Handler<AsyncResult<Void>> handler) {
-    throw new UnsupportedOperationException();
+    inAutoCommit.set(autoCommit);
+    if (inTransaction.compareAndSet(true, false) && autoCommit) {
```

Comment:
overall I believe we should avoid to use atomic here and use plain synchronization because we are checking the state of two variables and there can be race issues.